azure/modules/dns/dns.tf

```diff
@@ -0,0 +1,136 @@
+resource "azurerm_dns_zone" "cncf" {
+  name = "${ var.internal-tld }"
+  resource_group_name = "${ var.name }"
+}
+
+#Name Servers
+resource "null_resource" "dns_dig" {
+  # Error running plan: 1 error(s) occurred:
+  # * module.dns.null_resource.ns_to_ip_list: null_resource.ns_to_ip_list: value of 'count' cannot be computed
+  # FIXME: hardcoding to 2
+  # we only need 2 dns servers
+  count = 2
+  depends_on = [ "azurerm_dns_zone.cncf" ]
+  provisioner "local-exec" {
+    # filename = "/dev/null"
+    command = <<EOF
+# grab ip for this nameserver into a file
+dig +short ${ element(azurerm_dns_zone.cncf.name_servers,count.index) } > ${ var.name-servers-file }.${ count.index }.ip
+EOF
+  }
+}
+
+resource "null_resource" "dns_gen" {
+  depends_on = [ "null_resource.dns_dig" ]
+  provisioner "local-exec" {
+    # filename = "/dev/null"
+    command = <<EOF
+# collect them all into a csv
+# wait for them all to appear
+sleep 4
+cat ${ var.name-servers-file }.*.ip \
+| sed -n -e 'H;$${x;s/\n/,/g;s/^,//;p;}' \
+| tr -d '\n' \
+> ${ var.name-servers-file}
+EOF
+  }
+}
+
+resource "azurerm_dns_a_record" "A-etcd"  {
+  name = "etcd"
+  zone_name = "${azurerm_dns_zone.cncf.name}"
+  resource_group_name = "${ var.name }"
+  ttl = "300"
+  records = [
+    "${ var.master-ips }"
+  ]
+}
+
+resource "azurerm_dns_a_record" "A-etcds" {
+  count = "${ length(var.master-ips) }"
+
+  name = "etcd${ count.index+1 }"
+  zone_name = "${azurerm_dns_zone.cncf.name}"
+  resource_group_name = "${ var.name }"
+  ttl = "300"
+  records = [
+    "${ element(var.master-ips, count.index) }"
+  ]
+}
+
+resource "azurerm_dns_a_record" "A-master" {
+  name = "master"
+  zone_name = "${azurerm_dns_zone.cncf.name}"
+  resource_group_name = "${ var.name }"
+  ttl = "300"
+  records = [ "${ var.master-ips }" ]
+}
+
+resource "azurerm_dns_a_record" "A-masters" {
+  count = "${ length(var.master-ips) }"
+  name = "master${ count.index+1 }"
+  zone_name = "${azurerm_dns_zone.cncf.name}"
+  resource_group_name = "${ var.name }"
+  ttl = "300"
+  records = [
+    "${ element(var.master-ips, count.index) }"
+  ]
+}
+
+resource "azurerm_dns_srv_record" "etcd-client-tcp" {
```

Comment:
Using DNS for srv discovery should work across all providers.
This approach pulls the IP out of the VM regardless of vpc usage or not.
This was mainly to document the location in our code for hashicorp/terraform#13386